app/controllers/programs.js

```diff
@@ -21,40 +20,31 @@ export default Controller.extend({
   titleFilter: null,
 
   changeTitleFilter: task(function * (value) {
-    const clean = escapeRegExp(value);
-    this.set('titleFilter', clean);
+    this.set('titleFilter', value);
     yield timeout(250);
-
-    return clean;
+    return value;
   }).restartable(),
 
   hasMoreThanOneSchool: gt('model.schools.length', 1),
 
-  filteredPrograms: computed('changeTitleFilter.lastSuccessful.value', 'programs.[]', {
-    get() {
-      let defer = RSVP.defer();
-      let title = this.get('changeTitleFilter.lastSuccessful.value');
-      let exp = new RegExp(title, 'gi');
-
-      this.get('programs').then(programs => {
-        let filteredPrograms;
-
-        if(isEmpty(title)){
-          filteredPrograms = programs;
-        } else {
-          filteredPrograms = programs.filter(program => {
-            return isPresent(program.get('title')) && program.get('title').match(exp);
-          });
-        }
-
-        defer.resolve(filteredPrograms.sortBy('title'));
-      });
-
-      return PromiseArray.create({
-        promise: defer.promise
+  filteredPrograms: computed('changeTitleFilter.lastSuccessful.value', 'programs.[]', async function() {
```

Comment:
Same thing here. `changeTitleFilter.lastSuccessful.value` should be `titleFilter`.